File: MiniDOMBase/DMD/Sistema/Procedure/CProcedureCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;

namespace minidom
{
    public partial class Sistema
    {

        /// <summary>
        /// Cursore sulla tabella delle procedure
        /// </summary>
        [Serializable]
        public class CProcedureCursor 
            : minidom.Databases.DBObjectCursor<CProcedura>
        {
            private DBCursorStringField m_Tipo = new DBCursorStringField("Tipo");
            private DBCursorStringField m_Nome = new DBCursorStringField("Nome");
            private DBCursorField<ProceduraFlags> m_Flags = new DBCursorField<ProceduraFlags>("Flags");
            private DBCursorField<PriorityEnum> m_Priority = new DBCursorField<PriorityEnum>("Priorita");

            /// <summary>
            /// Costruttore
            /// </summary>
            public CProcedureCursor()
            {
            }

            /// <summary>
            /// Priority
            /// </summary>
            public DBCursorField<PriorityEnum> Priority
            {
                get
                {
                    return m_Priority;
                }
            }

            /// <summary>
            /// Tipo
            /// </summary>
            public DBCursorStringField Tipo
            {
                get
                {
                    return m_Tipo;
                }
            }

            /// <summary>
            /// Nome
            /// </summary>
            public DBCursorStringField Nome
            {
                get
                {
                    return m_Nome;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<ProceduraFlags> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Sistema.Procedure; //.Module;
            }


            //public override string GetTableName()
            //{
            //    return "tbl_CalendarProcs";
            //}

            //protected internal override CDBConnection GetConnection()
            //{
            //    return APPConn;
            //}
        }
    }
}

File: MiniDOMBase/DMD/Sistema/Procedure/CProcedureCursor.vb
Imports Microsoft.VisualBasic
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases

Partial Class Sistema


    <Serializable>
    Public Class CProcedureCursor
        Inherits DBObjectCursor(Of CProcedura)

        Private m_Tipo As New CCursorFieldObj(Of String)("Tipo")
        Private m_Nome As New CCursorFieldObj(Of String)("Nome")
        Private m_Flags As New CCursorField(Of ProceduraFlags)("Flags")
        Private m_Priority As New CCursorField(Of PriorityEnum)("Priorita")

        Public Sub New()

        End Sub

        Public ReadOnly Property Priority As CCursorField(Of PriorityEnum)
            Get
                Return Me.m_Priority
            End Get
        End Property

        Public ReadOnly Property Tipo As CCursorFieldObj(Of String)
            Get
                Return Me.m_Tipo
            End Get
        End Property

        Public ReadOnly Property Nome As CCursorFieldObj(Of String)
            Get
                Return Me.m_Nome
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of ProceduraFlags)
            Get
                Return Me.m_Flags
            End Get
        End Property


        Protected Overrides Function GetModule() As CModule
            Return Sistema.Procedure.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_CalendarProcs"
        End Function

        Protected Friend Overrides Function GetConnection() As CDBConnection
            Return Databases.APPConn
        End Function
    End Class

End Class

